library.js

```diff
@@ -145,141 +191,116 @@ plugin.verifyUser = function(uid, callback) {
 	});
 };
 
-plugin.findUser = function(payload, callback) {
-	// If payload id resolves to a user, return the uid, otherwise register a new user
-	winston.verbose('[session-sharing] Payload verified');
-
-	var parent = plugin.settings['payload:parent'],
-		id = parent ? payload[parent][plugin.settings['payload:id']] : payload[plugin.settings['payload:id']],
-		email = parent ? payload[parent][plugin.settings['payload:email']] : payload[plugin.settings['payload:email']],
-		username = parent ? payload[parent][plugin.settings['payload:username']] : payload[plugin.settings['payload:username']],
-		firstName = parent ? payload[parent][plugin.settings['payload:firstName']] : payload[plugin.settings['payload:firstName']],
-		lastName = parent ? payload[parent][plugin.settings['payload:lastName']] : payload[plugin.settings['payload:lastName']],
-		picture = parent ? payload[parent][plugin.settings['payload:picture']] : payload[plugin.settings['payload:picture']];
-
-	var fullname = [firstName, lastName].join(' ').trim();
-
-	if (!username && firstName && lastName) {
-		username = fullname;
-	} else if (!username && firstName && !lastName) {
-		username = firstName;
-	} else if (!username && !firstName && lastName) {
-		username = lastName;
+plugin.findOrCreateUser = function(userData, continueAfterFindOrCreate) {
```

Comment:
Ah, `continueAfterFindOrCreate` threw me for a loop here, I'm used to just naming it `callback`. Is there any particular reason you decided to name the callback something special for each step?